miro_preprocessor/image_sorter/src/sorter_logic.py

```diff
@@ -6,23 +6,206 @@
+
+    def _key_matches(self, key, string_list):
+        return any([self._compare(key, string) for string in string_list])
+
+    def _parse_date(self, date_string):
+        return dateutil.parser.parse(date_string, dayfirst=True)
+
+    def is_collection(self, *args):
+        return any([self._is_collection(collection) for collection in args])
+
+    @property
+    def image_library_dept_is_Archives_and_Manuscripts(self):
+        return self._compare("image_library_dept", "Archives and Manuscripts")
+
+    @property
+    def image_library_dept_is_Public_programmes(self):
+        return self._compare("image_library_dept", "Public programmes")
+
+    @property
+    def image_tech_captured_mode_is_videodisc(self):
+        return self._compare("image_tech_captured_mode", "videodisc")
+
+    @property
+    def is_innopac_id_8_digits(self):
+        return self._search(r"[0-9]{7}[0-9|x]{1}", "image_innopac_id") is not None
+
+    @property
+    def is_title_blank(self):
+        return self._is_blank("image_title")
+
+    @property
+    def is_title_empty(self):
+        return self.is_title_blank or self._key_matches('image_title', self._empty_title_strings)
+
+    @property
+    def is_image_pub_title_blank(self):
+        return self._is_blank("image_pub_title")
+
+    @property
+    def is_image_pub_periodical_blank(self):
+        return self._is_blank("image_pub_periodical")
+
+    @property
+    def is_after_first_march_2016(self):
+        scanned_date = self._get("image_tech_scanned_date")
+        if scanned_date is not None:
+            return self._parse_date(scanned_date) > self._parse_date("01/03/2016")
+        else:
+            return False
+
+    @property
+    def is_copyright_cleared(self):
+        return self._get('image_copyright_cleared') == 'Y'
+
+    @property
+    def is_cleared(self):
+        return self._get('image_cleared') == 'Y'
+
+    @property
+    def is_not_general_use(self):
+        return self._get('image_general_use') == 'N'
+
+    @property
+    def has_access_restrictions(self):
+        return not (self._is_blank("image_access_restrictions") or self._key_matches("image_access_restrictions",
+                                                                                     self._cc_accesses))
+
+    @property
+    def has_not_use_restrictions(self):
+        return not self._is_blank("image_use_restrictions") and self._key_matches("image_use_restrictions", self._cc_accesses)
+
+    @property
+    def is_not_for_public_access(self):
+        return (not self.is_copyright_cleared) or self.is_not_general_use or self.has_access_restrictions
+
+    @property
+    def is_for_public_access(self):
+        return not self.is_not_for_public_access
+
+    @property
+    def satisfies_api_filters(self):
+        return self.is_copyright_cleared and self.is_cleared and self.has_not_use_restrictions
+
+    @property
+    def is_cold_store(self):
+        return self.is_collection("F") or \
```

Comment:
nit: Python treats everything inside a set of parentheses as one line, so you don't need line continuations. This means you can write:

```python
return (
    self.is_collection("F") or
    (self.is_collection("L", "M", "V") and self.image_library_dept_is_Archives_and_Manuscripts) or
    ...
)
```

which I find slightly neater.

Up to you though, it makes no functional difference.